src/commands/node/upgrade.ts

```diff
@@ -0,0 +1,227 @@
+/*
+ * LiskHQ/lisk-commander
+ * Copyright © 2017–2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import { flags as flagParser } from '@oclif/command';
+import Axios from 'axios';
+import * as fsExtra from 'fs-extra';
+import Listr from 'listr';
+import semver from 'semver';
+import BaseCommand from '../../base';
+import { NETWORK, RELEASE_URL } from '../../utils/constants';
+import { download, extract, validateChecksum } from '../../utils/download';
+import { flags as commonFlags } from '../../utils/flags';
+import { liskTar, liskTarSHA256 } from '../../utils/node/commons';
+import { defaultBackupPath, getConfig } from '../../utils/node/config';
+import {
+	describeApplication,
+	Pm2Env,
+	registerApplication,
+	unRegisterApplication,
+} from '../../utils/node/pm2';
+import { getLatestVersion } from '../../utils/node/release';
+import { exec, ExecResult } from '../../utils/worker-process';
+import StartCommand from './start';
+import StopCommand from './stop';
+
+interface Flags {
+	readonly name: string;
+	readonly version: string;
+}
+
+interface PackageJson {
+	readonly version: string;
+}
+
+const getVersionToUpgrade = async (version: string, network: string) => {
```

Comment:
don't we have this on utils?

Reply:
no.. thats different, this fnction is to get the latest version if version parameter is undefined

Reply:
I think it can make the util function to be more generic.
Also, if it can be undefined, please use optional `version?: string` and optional should come at the end